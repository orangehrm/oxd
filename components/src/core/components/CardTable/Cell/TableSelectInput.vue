<template>
  <oxd-select-input
    up-icon="chevron-up"
    down-icon="chevron-down"
  ></oxd-select-input>
</template>

<script lang="ts">
import {defineComponent} from 'vue';
import SelectInput from '@orangehrm/oxd/core/components/Input/Select/SelectInput.vue';

export default defineComponent({
  name: 'oxd-table-select-input',
  components: {
    'oxd-select-input': SelectInput,
  },
});
</script>

<style src="./table-select-input.scss" lang="scss" scoped></style>
