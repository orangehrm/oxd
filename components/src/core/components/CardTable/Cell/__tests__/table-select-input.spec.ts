import {shallowMount} from '@vue/test-utils';
import TableSelectInput from '@orangehrm/oxd/core/components/CardTable/Cell/TableSelectInput.vue';

describe('TableSelectInput.vue', () => {
  it('should render TableSelectInput', async () => {
    const wrapper = shallowMount(TableSelectInput, {
      props: {
        options: [
          {id: 1, name: 'Option 1'},
          {id: 2, name: 'Option 2'},
        ],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
