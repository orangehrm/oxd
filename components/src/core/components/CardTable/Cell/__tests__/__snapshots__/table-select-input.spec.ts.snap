// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableSelectInput.vue should render TableSelectInput 1`] = `<oxd-select-input-stub options="[object Object],[object Object]" disabled="false" hidedropdowndefaultlabel="false" placeholder="-- Select --" dropdownposition="bottom" showemptyselector="true" up-icon="oxd-chevron-up" down-icon="oxd-chevron-down" dropdown-icon-size="xxx-small"></oxd-select-input-stub>`;
